src/sage/schemes/hodge_special_fourfolds/sff.py

```diff
@@ -0,0 +1,3099 @@
+        multi-rational map consisting of 2 rational maps
+        source variety: hypersurface in PP^5 defined by a form of degree 2
+        target variety: 4-dimensional subvariety of PP^4 x PP^5 cut out by 9 hypersurfaces of multi-degrees (0,2)^1 (1,1)^8
+        dominance: true
+        degree: 1
+        <BLANKLINE>
+        MultirationalMap (birational map from hypersurface in PP^5 to 4-dimensional subvariety of PP^4 x PP^5)
+
+    """
+    def __init__(self, X, Y, polys):
+        r"""See :class:`Rational_map_between_embedded_projective_varieties` for documentation."""
+        if Y is None:
+            if len(polys) > 0:
+                Y = PP(len(polys)-1, X.base_ring(), 'y')
+            else:
+                Y = ProjectiveSpace(0, X.base_ring(), 'empty')
+                Y = Embedded_projective_variety(Y,[Y.coordinate_ring().one()])
+                polys = [X.coordinate_ring().one()]
+        X = _check_type_embedded_projective_variety(X)
+        Y = _check_type_embedded_projective_variety(Y)
+        assert(isinstance(X,Embedded_projective_variety) and isinstance(Y,Embedded_projective_variety))
+        if X.base_ring() is not Y.base_ring():
+            raise ValueError("different base rings found")
+        if not isinstance(polys,tuple):
+            polys = tuple(polys)
+        if len(polys) != Y.ambient().dimension() + 1:
+            raise ValueError("got wrong number of polynomials")
+        H = X.Hom(Y)
+        gensR = matrix(polys).base_ring().gens()
+        if len(gensR) != X.ambient().coordinate_ring().ngens():
+            raise ValueError("expected polynomials in the coordinate ring of the source")
+        s = dict(zip(gensR, X.ambient().coordinate_ring().gens()))
+        polys = tuple([pol.subs(s) for pol in polys])
+        SchemeMorphism_polynomial_projective_space.__init__(self, H, polys)
+        assert(self.domain() is X)
+        assert(self.codomain() is Y)
+        # assert(self.defining_polynomials() == polys) # this fails
+        degs = [g.lift().degree() if hasattr(g,"lift") and (not hasattr(g,"degree")) else g.degree() for g in [pol for pol in polys if pol != 0]]
+        self.__degree_Forms = max(degs)
+        assert(self.__degree_Forms == min(degs))
+        self._source = X
+        self._target = Y
+        self._is_isomorphism = None
+        self._is_birational = None
+        self._is_dominant = True if self.target().dimension() < 0 else None
+        self._is_morphism = None
+
+    def _repr_(self):
+        r"""Return a string representation of the rational map
+
+        OUTPUT:
+
+        A string.
+
+        EXAMPLES::
+
+            sage: veronese(1,4)._repr_()
+            'rational map defined by forms of degree 4\nsource: PP^1\ntarget: PP^4'
+
+        """
+        if self.target().dimension() < 0:
+            return("empty rational map" + "\nsource: " + self.source()._repr_() + "\ntarget: " + self.target()._repr_())
```

Comment:
Morphisms in Sage usually print as "From: ... To:"

Reply:
Yes, but I'd leave it unchanged if it's not a problem. One of the reasons I defined this class is because I wanted to change how rational maps are printed. For example I never like to see the polynomials that define it.